gui/webdisplay/src/RWebWindow.cxx

```diff
@@ -1811,3 +1984,60 @@ bool RWebWindow::EmbedFileDialog(const std::shared_ptr<RWebWindow> &window, unsi
 
    return gStartDialogFunc(window, connid, args);
 }
+
+/////////////////////////////////////////////////////////////////////////////////////
+/// Calculate HMAC checksum for provided key and message
+/// Key combained from connection key and session key
+
+std::string RWebWindow::HMAC(const std::string &key, const std::string &sessionKey, const char *msg, int msglen)
+{
+   using namespace ROOT::Internal::SHA256;
+
+   auto get_digest = [](sha256_t &hash, bool as_hex = false) -> std::string {
+      std::string digest;
+      digest.resize(32);
+
+      sha256_final(&hash, reinterpret_cast<unsigned char *>(digest.data()));
+
+      if (!as_hex) return digest;
+
+      static const char* digits = "0123456789abcdef";
+      std::string hex;
+      for (int n = 0; n < 32; n++) {
+         unsigned char code = (unsigned char) digest[n];
+         hex += digits[code / 16];
+         hex += digits[code % 16];
+      }
+      return hex;
+   };
+
+   // calculate hash of sessionKey + key;
+   sha256_t hash1;
+   sha256_init(&hash1);
+   sha256_update(&hash1, (const unsigned char *) sessionKey.data(), sessionKey.length());
+   sha256_update(&hash1, (const unsigned char *) key.data(), key.length());
+   std::string kbis = get_digest(hash1);
```

Comment:
See above, I believe kbis needs to be padded to the 64 byte block size.

Reply:
Yes, 64 byte padding will be provided.